scripts/rimpl.ts

```diff
@@ -0,0 +1,72 @@
+import {parseProcessArgument} from '../src/util/args';
+import {openYamlFileAsObject, saveYamlFileAs} from '../src/util/yaml';
+import {Observatory} from '../src/util/observatory';
+import {ModelsUniverse} from '../src/util/models-universe';
+
+/**
+ * For each graph builds params, then passes it to computing fn.
+ */
+const calculateImpactsBasedOnGraph =
+  (impl: any, modelsHandbook: ModelsUniverse) => async (service: string) => {
+    const serviceData = impl.graph.children[service];
+    const {pipeline, observations, config} = serviceData;
+
+    const observatory = new Observatory(observations);
+
+    for (const model of pipeline) {
+      const instance: any = await modelsHandbook.initalizedModels[model](
+        config[model]
+      );
+
+      await observatory.doInvestigationsWith(instance);
+    }
+
+    const impact = observatory.getObservedImpact();
+    impl.graph.children[service].impact = impact;
+
+    return impl;
+  };
+
+/**
+ * 1. Parses yml input/output process arguments.
+ * 2. Opens yaml file as an object.
+ * 3. Initializes models.
+ * 4. Initializes graph, does computing.
+ * 5. Saves processed object as a yaml file.
+ * @example run following command `npx ts-node scripts/rimpl.ts --impl ./test.yml --ompl ./result.yml`
+ */
+const rimplPOCScript = async () => {
+  try {
+    const {inputPath, outputPath} = parseProcessArgument();
+    const impl = await openYamlFileAsObject(inputPath);
+
+    if (!('graph' in impl)) {
+      throw new Error('No graph data found.');
+    }
+
+    // Lifecycle Initialize Models
+    const modelsHandbook = new ModelsUniverse();
+
+    impl.initialize.models.forEach((model: any) =>
+      modelsHandbook.writeDown(model)
+    );
+
+    // Initialize impact graph/computing
+    const services = Object.keys(impl.graph.children);
```

Comment:
@narekhovhannisyan FYI for PoC this is fine for now but it eventually needs to be something that parses the children, the children of children, etc.... and collects all the component nodes (there are grouping nodes and component nodes, component nodes are the nodes that hold observations). Component nodes will never have children.

Then we loop through the component nodes.

The name of the list of things should be "components" or "componentNodes" I would not call these services.

Reply:
yes we may need recursive approach, this flat version just for having working structure